### ISSUE TITLE
Getter with polymorphic type

### ISSUE THREAD

**Issue by mhluska** [mrichar1/jsonapi-vuex#25]
I have a need to look up an item by id which has a polymorphic type. For example:

```js
this.$store.getters['jv/get']({ _jv: { type: 'card' } })[this.itemId]
```

But this will fail because the item types returned by my JSON API are actually `credit-card` and `debit-card`. As an ugly hack, I can call this function repeatedly for all the possible types until one resolves with the data.

Would it make sense to have a getter that does not require specifying a type since IDs should be unique across types in the database?

**Comment by mrichar1** [comment]
I think the right way to go about handling records from the store with globally unique IDs is to use the jsonpath functionality provided in the `get` getter. Bear in mind that not all APIs have globally unique ids, and that ids muct only be unique for each type:

> Within a given API, each resource object’s `type` and `id` *pair* **MUST** identify a single, unique resource.  

https://jsonapi.org/format/#document-resource-object-identification

You should be able to specify no type or id, which would normally return the whole store, and then filter by id - so to get all resources with `id = 10` you'd do:

`this.$store.getters['jv/get']('', '$.*.10')`

As it turned out, this wasn't previously possible, since the code path for returning the whole store missed the `jsonpath` filtering code, but I've fixed this in `0.1.1`

**Comment by mhluska** [comment]
Aah perfect. This explains a lot. Thanks.
